src/schema/types.rs

```diff
@@ -74,6 +74,45 @@ impl Type {
     }
   }
 
+  /// Check if 'sub_type' schema is part of current schema, e.g. projected columns
+  pub fn check_contains(&self, sub_type: &Type) -> bool {
+    // names match, and repetitions match or not set for both
+    let basic_match = self.get_basic_info().name() == sub_type.get_basic_info().name() && (
+      self.is_schema() && sub_type.is_schema() || !self.is_schema() && !sub_type.is_schema() &&
+      self.get_basic_info().repetition() == sub_type.get_basic_info().repetition()
+    );
+
+    match *self {
+      Type::PrimitiveType { .. } if basic_match && sub_type.is_primitive() => {
+        self.get_physical_type() == sub_type.get_physical_type()
+      },
+      Type::GroupType { .. } if basic_match && sub_type.is_group() => {
+        // build hashmap of name -> TypePtr
+        let mut field_index = HashMap::new();
+        for field in self.get_fields() {
+          field_index.insert(field.name(), field);
+        }
+
+        let mut found_subtype = true;
+        for field in sub_type.get_fields() {
+          found_subtype = found_subtype && match field_index.get(field.name()) {
```

Comment:
nit: we can quit early if `found_subtype` is false.

Reply:
Will update, thanks.